internal/controller/datamovementmanager_controller.go

```diff
@@ -329,26 +337,92 @@ func (r *DataMovementManagerReconciler) updateLustreFileSystemsIfNecessary(ctx c
 	for _, lustre := range filesystems.Items {
 		_, found := lustre.Spec.Namespaces[manager.Namespace]
 		if !found {
-
 			if lustre.Spec.Namespaces == nil {
 				lustre.Spec.Namespaces = make(map[string]lusv1beta1.LustreFileSystemNamespaceSpec)
 			}
 
 			lustre.Spec.Namespaces[manager.Namespace] = lusv1beta1.LustreFileSystemNamespaceSpec{
 				Modes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany},
 			}
+		}
 
-			if err := r.Update(ctx, &lustre); err != nil {
-				return err
-			}
+		// Add the dm finalizer to keep this resource from being deleted until dm is no longer using it
+		if lustre.DeletionTimestamp.IsZero() && !controllerutil.ContainsFinalizer(&lustre, finalizer) {
+			controllerutil.AddFinalizer(&lustre, finalizer)
+		}
 
-			log.Info("Updated LustreFileSystem", "object", client.ObjectKeyFromObject(&lustre).String(), "namespace", manager.Namespace)
+		if err := r.Update(ctx, &lustre); err != nil {
+			return err
 		}
+		log.Info("Updated LustreFileSystem", "object", client.ObjectKeyFromObject(&lustre).String(), "namespace", manager.Namespace)
 	}
 
 	return nil
 }
 
+func (r *DataMovementManagerReconciler) removeLustreFileSystemsFinalizersIfNecessary(ctx context.Context, manager *dmv1alpha1.DataMovementManager) (*ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	filesystems := &lusv1beta1.LustreFileSystemList{}
+	if err := r.List(ctx, filesystems); err != nil && !meta.IsNoMatchError(err) {
+		return nil, fmt.Errorf("list lustre file systems failed: %w", err)
+	}
+
+	// Get the DS to compare the list of volumes
+	ds := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      daemonsetName,
+			Namespace: manager.Namespace,
+		},
+	}
+	if err := r.Get(ctx, client.ObjectKeyFromObject(ds), ds); err != nil {
+		return nil, err
+	}
+
+	finalizersToRemove := []lusv1beta1.LustreFileSystem{}
+	for _, lustre := range filesystems.Items {
+		// This lustre is in the process of deleting, verify it is not in the list of DS volumes
+		if !lustre.DeletionTimestamp.IsZero() {
+			finalizersToRemove = append(finalizersToRemove, lustre)
+			for _, vol := range ds.Spec.Template.Spec.Volumes {
+				if lustre.Name == vol.Name {
+					log.Info("Requeue: wait for daemonset to drop lustrefilesystem volume", "lustrefilesystem", lustre)
+					return &ctrl.Result{RequeueAfter: 1 * time.Second}, nil
```

Comment:
This controller has an `Owns(&appsv1.DaemonSet{})`, so the requeue shouldn't be necessary.